File: src/jaslib_src/jaslib/language_services/janome_ex/word_extraction/matches/state_tests/another_match_owns_the_form.py
# from __future__ import annotations
#
# from typing import TYPE_CHECKING
#
# from autoslot import Slots
#
# from jaslib.language_services.janome_ex.word_extraction.matches.state_tests.head.failed_match_requirement import FailedMatchRequirement
#
# if TYPE_CHECKING:
#     from jaslib.language_services.janome_ex.word_extraction.matches.requirements.vocab_match_inspector import VocabMatchInspector
#
# class ForbidsAnotherMatchIsHigherPriority(Slots):
#     _failed: FailedMatchRequirement = FailedMatchRequirement.forbids("another_match_owns_the_form_or_is_higher_priority_due_to_custom_requirements_weight")
#
#     @classmethod
#     def apply_to(cls, inspector: VocabMatchInspector) -> FailedMatchRequirement | None:
#         if inspector.match.another_match_is_higher_priority:
#             return cls._failed
#         return None

File: src/src_dotnet/JAStudio.Core/LanguageServices/JanomeEx/WordExtraction/Matches/StateTests/AnotherMatchOwnsTheForm.cs
using JAStudio.Core.LanguageServices.JanomeEx.WordExtraction.Matches.Requirements;
using JAStudio.Core.LanguageServices.JanomeEx.WordExtraction.Matches.StateTests.Head;

namespace JAStudio.Core.LanguageServices.JanomeEx.WordExtraction.Matches.StateTests;

static class ForbidsAnotherMatchIsHigherPriority
{
   static readonly FailedMatchRequirement Failed = FailedMatchRequirement.Forbids("another_match_owns_the_form_or_is_higher_priority_due_to_custom_requirements_weight");

   public static FailedMatchRequirement? ApplyTo(VocabMatchInspector inspector)
   {
      if(inspector.Match.AnotherMatchIsHigherPriority)
      {
         return Failed;
      }

      return null;
   }
}
